docs/architecture/adr-0003-mvvm-architecture.md

```diff
@@ -0,0 +1,37 @@
+# ADR 3: Refactoring into a Model/View/View-Model (MVVM) architecture
+
+## Context
+The initial version of the codebase was built without a specific architecture in mind, and this has led to a few problems:
+1. New UX flow proposals were incompatible with previous assumptions of how different views were bound together, and those were fragile and difficult to change.
+1. Integrating large pieces of the android-components library caused many regressions, partly because state management is distributed throughout different parts of the UI.
+1. Testing app state (especially the BrowserOverlay, or homescreen) was difficult because side effects of actions were spread out throughout the app, rather than being consolidated.
+1. A formal architecture was needed to provide clarity for where to change/add code as the number of people working with the codebase grows.
+
+Formal architectures address these problems by explicitly decoupling the state and UI. The common ones are MVC/MVP, MVVM, MVI.
+
+## Decision
+We decided on MVVM because in addition to satisfying the other requirements of separating state and views, it also allowed for incremental change, which was important because switching to a new architecture would extend past a single sprint, and there was the possibility that we’d need to respond to critical issues during the refactor. It’s also a familiar and well-documented architecture, and integrates well with Android’s `ViewModel` and `LiveData`.
```

Comment:
nit: maybe mention we're doing Google's MVVM with Architecture components and link to it: https://developer.android.com/jetpack/docs/guide which is why it's well documented and familiar.